### DISCUSSION TITLE
Wemod won't start

### DISCUSSION THREAD

**Discussion by mzigaib** [DeckCheatz/wemod-launcher#137] [Q&A] [answered]
I was using wemod and it was working until I did a fresh install on my Fedora 41 and I followed all the instructions to install it again I am using Lutris with the prefix command which is kind of working, I can see the terminal launching and the game starts but wemod does not launch as before, any tips?

**Comment by mzigaib** [comment]
I would really appreciate some help here
I just see this on the shell but wemod for some reason does not start
![image](https://github.com/user-attachments/assets/0659ebe2-36a3-46ed-810c-a137aafa54c5)

Thanks!

**Reply by marvin1099** [reply]
Is this still a issue, it says in the terminal that wemod was found with pid 352,
this suggests it is running in the background,
it may start as a tray service.
try to delete the wemod_data folder,
this will reset the accounts and settings of wemod.
After that it should open (you will have to re-login). 
You can also try to delete the WeMod.exe in the folder wemod_bin,
this should re-download wemod and may also fix this.

**Comment by mzigaib** [comment]
Hey thanks for the feedback
I did what you said about deleting the data folder and the .exe
I still get the same issue where wemod does not launch.
The weird thing is that I am sure that I configure the same way as before when it was working.
![Screenshot From 2024-12-02 14-51-35_edited](https://github.com/user-attachments/assets/a55ac8be-e8c8-48a6-a69d-3f1c950a59fd)


Some couple of notes:
If I run through bottles wemod starts but I get other issues with the game like some videos not being able to play error, so maybe it is something to do with my fresh Lutris install?

**Reply by marvin1099** [reply]
Maybe try other a other wine version.
By default lutris uses the system wine version I think.
I got it to work with wine GE and proton ge.
They can both be downloaded / selected in lutris.

**Comment by mzigaib** [comment]
Ok I will try that.

**Comment by mzigaib** [comment]
I still have the same issue
I did try to:
-Re-install Lutris completely
-Delete game pre-fix and wemod.exe
-I tried to use  different version of wine on Lutris like wine_ge_8-26,27 and also wine-ge_proton-26

Any other thips?

**Reply by marvin1099** [reply]
Try a proton version,
should be in lutris,
but steam may have to be installed for proton

**Reply by mzigaib** [reply]
Ok let me see that.

**Comment by mzigaib** [comment]
This is what I have on Lutris I tried those installed 
![image](https://github.com/user-attachments/assets/7e0044e8-6a97-4a20-bbfc-78945c038bc2)

**Reply by marvin1099** [reply]
Try to run:
```
cd /home/$USER/wemod-launcher
git reset --hard origin
git pull
sudo chmod -R ug+x .
```
And retry

**Comment by mzigaib** [comment]
Done but I still got the same issue, I know it is something on my system.
![image](https://github.com/user-attachments/assets/363cce26-0377-48f6-9b08-e4c7e4be0024)

**Reply by marvin1099** [reply]
Upload the wemod.log right here as a file.
I like to check the output.

**Comment by mzigaib** [comment]
Sure!
[wemod.log](https://github.com/user-attachments/files/17983276/wemod.log)

**Reply by marvin1099** [reply]
Try to run something like
```
WINE="Path-to-game-wine-prefix" "Path-to-lutris-wine-ge" "Path-to-Wemod.exe"
```
And see if this starts wemod

**Reply by mzigaib** [reply]
Like this?
WINE="/mnt/DRIVE_D/Lutris" "/home/user/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64"  "/home/user/wemod-launcher/wemod"

**Reply by marvin1099** [reply]
The wemod.exe file is in the wemod_bin folder

**Reply by marvin1099** [reply]
And also this path 
/home/user/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64
Should end with something like /wine
May still be correct but check if this is is infact the wine runner file (not a folder)

**Reply by marvin1099** [reply]
The command is probably like this:
```
WINE="/mnt/DRIVE_D/Lutris" "/home/user/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine" "/home/user/wemod-launcher/wemod_bin/WeMod.exe"
```

**Reply by mzigaib** [reply]
Does not even launch the game

**Reply by marvin1099** [reply]
This is only supposed to run WeMod.exe not the game, is that working?

**Reply by mzigaib** [reply]
That is the way I used
![image](https://github.com/user-attachments/assets/e4ea5962-e63d-4067-936e-f3fcc7c32103)

**Reply by mzigaib** [reply]
Running through terminal
![image](https://github.com/user-attachments/assets/6103803b-167b-453b-8dcb-9ea5d131a9e0)

**Reply by mzigaib** [reply]
> This is only supposed to run WeMod.exe not the game, is that working?

Sorry, it does not.

**Reply by mzigaib** [reply]
Through bottles I got it running.
I mean start Wemod not the game.

**Comment by marvin1099** [comment]
If this won't run Wemod the script can't do it.
Run it in the terminal, not over lutris.

**Reply by marvin1099** [reply]
This is a terminal but I like you to open a new terminal and just run that in there.
In the one you provided lutris still seems to be running but for the command you do not need to run lutris.

**Reply by marvin1099** [reply]
Of you are good with using bottels then say so.
I would close this discussion in that case.

**Reply by marvin1099** [reply]
We can of course also try to solve this.
Just give a quick awnser on if you like to try to fix this or if you are content (then I will close the  discussion)

**Reply by mzigaib** [reply]
I can just start wemod on bottles but I prefer to keep it on lutris, I don't know how to run prefix command there.
I am out now soon as I get back I will try to run the commands directly from the shell which I was not doing it correctly.

**Reply by marvin1099** [reply]
OK then go ahead and run this in a new terminal window:
```
WINE="/mnt/DRIVE_D/Lutris" "/home/user/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine" "/home/user/wemod-launcher/wemod_bin/WeMod.exe"
```

**Comment by mzigaib** [comment]
This is what I get when I run the command and it got stuck in there:
wineserver: using server-side synchronization.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
002c:fixme:winediag:LdrInitializeThunk wine-staging 8.0 is a testing version containing experimental patches.
002c:fixme:winediag:LdrInitializeThunk Please mention your exact version when filing bug reports on winehq.org.
007c:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
007c:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
007c:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
007c:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0090:err:hid:udev_bus_init UDEV monitor creation failed
0114:err:ole:apartment_add_dll couldn't find function DllGetClassObject in L"C:\\windows\\system32\\uiautomationcore.dll"
0114:err:ole:com_get_class_object no class object {ff48dba4-60ef-4201-aa87-54103eef594e} could be created for context 0x1
0114:err:tabtip:create_uia_event_handler Failed to create IUIAutomation interface, hr 0x800401f8
011c:fixme:powrprof:PowerRegisterSuspendResumeNotification (0x00000002,000000000081FE10,000000000081FE08) stub!
011c:err:sync:RtlpWaitForCriticalSection section 000000017006E620 "../dlls/ntdll/loader.c: loader_section" wait timed out in thread 011c, blocked by 0120, retrying (60 sec)

Maybe a firewall issue?

**Reply by marvin1099** [reply]
Hmm so there m not a wine expert but let's try a few things.
Firstly download wine, winetricks, winecfg
Some other packages may also help but only know these from the top of my head.
It may help to have these packages globally.

And see if you can then use
```
WINE="/mnt/DRIVE_D/Lutris" "/home/user/.local/share/lutris/runners/wine/wine-ge-8-26-x86_64/bin/wine" "/home/user/wemod-launcher/wemod_bin/WeMod.exe"
```

Next there are some lutris settings you can change related to wine you should try to activate them (dxvk for example is in there but I also dont remember all of them, but these can also be downloaded globally).
Try that with the command prefix settings from from the external launcher guide.

If you use wayland that may result in issues, try to run it with x11 or xwayland and see if that works.
If you dont know if you have Wayland of x11 just do a internet search on how to check and a internet search on how to switch to x11 or run apps with xwayland. 
I just don't use Wayland so I don't know this you have to figure that out yourself.

**Comment by mzigaib** [comment]
OK, I have all those installed and this is what I have on my Lustris setted
![image](https://github.com/user-attachments/assets/6a428946-f2ab-4cd0-9448-4a043f24f7e7)

And I am not using Wayland.

**Reply by marvin1099** [reply]
That's all I have really, I will go to sleep now.
We can continue tomorrow but I have no idea what is going on. I don't think  I can solve this but you can ask me some more stuff then I may stumble on a awnser, but tomorrow as I was saying I will go to sleep now.

**Reply by marvin1099** [reply]
I can recommend the wiki there are some issue resolving steps but I made the wiki so your luck may run out soon but you can try.
The full guide and the help pages may have something that will help you.
https://github.com/DeckCheatz/wemod-launcher/wiki

**Reply by mzigaib** [reply]
Off course man!
Thank you very much for you time and patience I really appreciate it.
Well, let me know if you happen to have any more ideas.
Have a good rest.

**Comment by mzigaib** [comment]
Hey me again,
I was hoping to get it working on bottles but I am getting some errors from Wemod.
Here is what I did

I create a bottle with a preset "Gaming" and set the "Launch options" exactly how the instructions point with the "Wemod dir" %command%

But I am getting this error on Wemod start on the log

The script wemod-launcher is running on version 1.509
No updates available
The subsplit args list is 1 long and the full contents are:
	[['/usr/bin/wine64', '/mnt/Seagate_10T/Hydra/Nioh 2 Complete Edition v.1.28.00 (2021)/Nioh 2 Complete Edition/nioh2.exe']]
ERR:
list index out of range

Any tips?

**Comment by marvin1099** [comment]
Try this;
```
WINEPREFIX=$YOUR-WINE-PREFIX-FOLDER-HERE WINE=$YOUR/WINE/PATH /home/$USER/Apps/Games/wemod-launcher/wemod $PATH/TO/GAME/EXE
```

bottles and lutris du not support "%command%".
That is a steam thing.
If you had "%command%" in your lutris start that must be replaced by the game path.
I that case you can retry lutris.
For bottles works basically the same.

**Reply by marvin1099** [reply]
Replace all the paths to the correct paths
All of the paths are placeholders or incorrect.

**Reply by marvin1099** [reply, marked as answer]
Not that I read this in lutris you dont need the to add the game path to the pree command.
Not shure what you tryed but the setup of lutris is described here 
https://github.com/DeckCheatz/wemod-launcher/wiki/Using-External-Launchers
If you like to retry that.
Always make shure the game can run without wemod before adding the wemod starter lines.

**Reply by mzigaib** [reply]
That is the way it was working before I made a fresh re-install of the system but now it is not and I don´t know why.
But thanks for the tip.

**Comment by mzigaib** [comment]
Got it!
I will try soon as I can!
Thanks man!
